project-management-infrastructure/src/main/java/life/qbic/projectmanagement/infrastructure/sample/SampleRepositoryImpl.java

```diff
@@ -72,6 +76,37 @@ public Result<Collection<Sample>, ResponseCode> addAll(Project project,
     return Result.fromValue(samples);
   }
 
+  @Transactional
+  @Override
+  public Result<Collection<Sample>, ResponseCode> deleteAll(Project project,
+      Collection<Sample> samples) {
+    Collection<SampleId> sampleIds = new ArrayList<>();
+    samples.forEach(sample -> sampleIds.add(sample.sampleId()));
```

Comment:
You should not modify a list in a foreach loop.
Please replace by an assignment.
```java
samples.stream().map(Sample::sampleId).toList();
```

Reply:
changed it anyway :)